src/RCDEF.CPP

```diff
@@ -1952,7 +1954,11 @@ nexTokRec: ;                            // come here after *word or error */
 			if (gtoks("s"))                             // read file name
 				rcderr("Bad name after *file.");
 			rchFileNm = stashSval(0);                   // store name for rec type definition and for have-file check below
-			sprintf( dbuff, "%s\\%sx", incdir, rchFileNm);
+			char rchFileNmX[FILENAME_MAX];				// rchFileNm variable with a x at the end
+			strcpy(rchFileNmX, rchFileNm);
+			rchFileNmX[strlen(rchFileNm)] = 'x';		// Add x
+			rchFileNmX[strlen(rchFileNm) + 1] = '\0';	// End of string character
+			xfcreatepath(incdir, rchFileNmX, dbuff);
```

Comment:
This feels like a convoluted way to simply add an "x" to the end of a string. What's wrong with `sprintf`?

Reply:
Nothing. Perhaps I've spent too much time in low C 😆. And because lower in the source file that's how he removes the x.